### PR TITLE
IOSF-320: Choosing Static Paths

### PR DESCRIPTION
**Pull request by mcfearsome**
## What's the purpose of this pull request?
Add documentation around the process of customizing the static paths

## How it works? 
Analog Optical Character Recognition

## How to test it?
Dynamic Organic Processing Unit

## References
n/a

### CODE REVIEW

**Comment by mcfearsome** [comment]
@iaronaraujo & @rafaelbac, I was unsure who I should add to review this so I am adding you 2 for guidance.

File: docs/static-paths.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+# Choosing Which Pages are Pre-Rendered
+
+One of the core principles of [JAMStack](https://github.com/vtex/faststore/tree/master/docs/what-is-jamstack.md) is the pre-rendering, or static generation, of pages at build time. This provides artifacts of highly optimized static pages, reducing the runtime complexity while increasing performance and resiliency. In the end were left with a super stable & faststore...
```
Comment (eric-pigeon) [inline docs/static-paths.md:3]: we're left with a

**Review by eric-pigeon — CHANGES_REQUESTED** [review]

**Review by eric-pigeon — APPROVED** [review]

**Comment by mcfearsome** [comment]
> Hey guys, what do we need to do to move this forward?

Hey @tlgimenes, sorry I have been out on PTO so haven't been available to bug anyone to actually review the content and approve it! I'd happily accept a review if you have the time.

**Review by iaronaraujo — APPROVED** [review]

**Review by viniagostini — APPROVED** [review]

File: docs/static-paths.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
+# Choosing Which Pages are Pre-Rendered
+
+One of the core principles of [JAMStack](https://github.com/vtex/faststore/tree/master/docs/what-is-jamstack.md) is the pre-rendering, or static generation, of pages at build time. This provides artifacts of highly optimized static pages, reducing the runtime complexity while increasing performance and resiliency. In the end we're left with a super stable & faststore...
+
+## Using the `@vtex/gatsby-vtex-source` Plugin
+
+>ℹ️ *We are assuming you have followed the [Getting Started](https://github.com/vtex/faststore/tree/master/docs/getting-started.mb) document and have a running store.*
```
Comment (lariciamota) [inline docs/static-paths.md:7]:
```suggestion
>ℹ️ *We are assuming you have followed the [Getting Started](https://github.com/vtex/faststore/tree/master/docs/getting-started.md) document and have a running store.*
```

File: docs/static-paths.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+### Overriding the Base Configuration
+
+If you want to take full control of the logic for deciding what resources are pre-rendered, you can do so by using the `getStaticPaths` Plugin Option adhering to the signature `() => Promise<string[]>`.
+**Best Practice** is to define a file, `getStaticPaths.js`, in the Project Root. This file should export a single async function and ultimately return an array of strings, or URL Paths if we want to be specific. The contents of this function are up to you but the [default function](https://github.com/vtex/faststore/blob/master/packages/gatsby-source-vtex/src/staticPaths.ts#L38) can be used as a blueprint for your customizations.
+This file would then be utilized in your projects `gatsby-config.js` and required so it can be used in the plugin configuration.
+```javascript
+# where getStaticPaths.js exports a single function
+# that returns an array of strings. Note that no options
+# are sent to the getStaticPaths function when used over
+# the default static paths function.
```
Comment (lariciamota) [inline docs/static-paths.md:42]:
```suggestion
// where getStaticPaths.js exports a single function
// that returns an array of strings. Note that no options
// are sent to the getStaticPaths function when used over
// the default static paths function.
```

File: docs/static-paths.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+| workspace      | string   | true      | n/a                                                                                                                |                                                                                                     |
+| environment    | pattern  | true      | n/a                                                                                                                | `(vtexcommercestable\|vtexcommercebeta)`                                                            |
+| getStaticPaths | function | false     | [staticPaths.ts](https://github.com/vtex/faststore/blob/master/packages/gatsby-source-vtex/src/staticPaths.ts#L38) | *Primary Subject of this Document*                                                                  |
+| getRedirects   | function | false     | n/a                                                                                                                | See [Managing URL Redirects](https://github.com/vtex/faststore/tree/master/docs/getting-started.mb) |
```
Comment (lariciamota) [inline docs/static-paths.md:69]:
```suggestion
| getRedirects   | function | false     | n/a                                                                                                                | See [Managing URL Redirects](https://github.com/vtex/faststore/tree/master/docs/managing-url-redirects.md) |
```

File: docs/static-paths.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+| pageTypes      | string[] | false     | `['Product', 'Department', 'Category', 'Brand', 'SubCategory',]`                                                   |                                                                                                     |
+| maxNumPaths    | number   | false     | 100                                                                                                                |                                                                                                     |
+| ignorePaths    | string[] | false     | n/a                                                                                                                | *Can be used to exclude paths from the final results returned by `getStaticPaths`                   |
```
Comment (lariciamota) [inline docs/static-paths.md:72]:
```suggestion
| ignorePaths    | string[] | false     | n/a                                                                                                                | Can be used to exclude paths from the final results returned by `getStaticPaths`                   |
```

File: docs/static-paths.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+| environment    | pattern  | true      | n/a                                                                                                                | `(vtexcommercestable\|vtexcommercebeta)`                                                            |
+| getStaticPaths | function | false     | [staticPaths.ts](https://github.com/vtex/faststore/blob/master/packages/gatsby-source-vtex/src/staticPaths.ts#L38) | *Primary Subject of this Document*                                                                  |
+| getRedirects   | function | false     | n/a                                                                                                                | See [Managing URL Redirects](https://github.com/vtex/faststore/tree/master/docs/getting-started.mb) |
+| pageTypes      | string[] | false     | `['Product', 'Department', 'Category', 'Brand', 'SubCategory',]`                                                   |                                                                                                     |
```
Comment (lariciamota) [inline docs/static-paths.md:70]:
```suggestion
| pageTypes      | string[] | false     | `['Product', 'Department', 'Category', 'Brand', 'SubCategory']`                                                   |                                                                                                     |
```

**Review by lariciamota — CHANGES_REQUESTED** [review]
Good job!
It'd also be nice to have a review from a tech writer, I believe @Mariana-Caetano can help us with that.

**Review by Mariana-Caetano — APPROVED** [review]

**Review by lariciamota — APPROVED** [review]

**Pull request merged**